### DISCUSSION TITLE
Detect Metamask Extension using standalone implementation

### DISCUSSION THREAD

**Discussion by ntoine223** [WalletConnect/walletconnect-monorepo#2142] [Developer Support]
Hello,

I am currently trying to configure my WalletConnect modal with the [standalone](https://docs.walletconnect.com/2.0/web3modal/standalone/installation) configuration. However, I can't get the modal to detect the presence of a Metamask extension.

I have tried other implementation methods (React/Htmk) and in these cases the MetaMask extension is detected. I concluded that the use of wagmi could affect this.

How can I detect the presence of the Metamask extension while staying in the standalone implementation and without using wagmi?

Thanks a lot.

**Comment by nunoplima** [comment]
Have a look here: https://docs.walletconnect.com/2.0/web3modal/about#versioning
"If the wallet you intend to use still utilizes v1, we recommend including version: 1 in your configuration to ensure that your users can continue to connect to their preferred wallet", this is the case of MM, not sure on how to include version 1 in the configuration should be done...

**Reply by ntoine223** [reply]
Hi, yes I have read that part in particular.
However, when I test online the implementation of version 2, here it is indeed about a detection of the walletconnect extension on desktop: https://web3modal-dev.pages.dev/v2Base

**Reply by aliburzhi** [reply]
Hello !
 i have got the same issue.
  did you solved it ? 
  when im trying to configure my wallet connect modal with the standalone i can't see metamask in wallet`s list
 i can see metamask after i clicked on your link and connect wallet btw 
![Screenshot 2023-04-06 at 23 05 12](https://user-images.githubusercontent.com/62848367/230483333-c61906a0-a2e0-4747-b7f8-bce5f5c99d7f.png)

**Reply by glitch-txs** [reply]
Standalone version doesn't work with injected providers like MetaMask. Only web3modal does. If you're using standalone version you'll need to handle MetaMask on your own separately.

**Reply by ntoine223** [reply]
Ok, thank you for this detail.

**Reply by aliburzhi** [reply]
Thanks
